src/core/server/elasticsearch/retry_call_cluster.ts

```diff
@@ -23,6 +23,53 @@ import * as legacyElasticsearch from 'elasticsearch';
 
 import { CallAPIOptions } from '.';
 
+const esErrors = legacyElasticsearch.errors;
+
+/**
+ * Retries the provided Elasticsearch API call when an error such as
+ * `AuthenticationException` `NoConnections`, `ConnectionFault`,
+ * `ServiceUnavailable` or `RequestTimeout` are encountered. The API call will
+ * be retried once a second, indefinitely, until a successful response or a
+ * different error is received.
+ *
+ * @param apiCaller
+ */
+
+// TODO: Replace with APICaller from './scoped_cluster_client' once #46668 is merged
+export function migrationsRetryCallCluster(
+  apiCaller: (
+    endpoint: string,
+    clientParams: Record<string, any>,
+    options?: CallAPIOptions
+  ) => Promise<any>
+) {
+  return (endpoint: string, clientParams: Record<string, any> = {}, options?: CallAPIOptions) => {
+    return defer(() => apiCaller(endpoint, clientParams, options))
+      .pipe(
+        retryWhen(errors =>
```

Comment:
I believe the previous behavior would log the error message on each poll loop.

Reply:
I think it would be useful to add some warning logs, but this would be new behaviour.

We logged version check failures (but only once for each unique set of node/version combinations). Authentication and other failures were silently swallowed by the health check. The x-pack licensing plugin didn't wait for ES to go green, so that periodically logged authentication failures. Since we block plugins from starting that will no longer provide a clue.